src/main/java/slimeknights/tconstruct/tables/client/inventory/widget/TinkerStationButtonsWidget.java

```diff
@@ -0,0 +1,75 @@
+package slimeknights.tconstruct.tables.client.inventory.widget;
+
+import com.google.common.collect.Lists;
+import net.minecraft.client.gui.components.Button;
+import slimeknights.tconstruct.library.client.Icons;
+import slimeknights.tconstruct.library.tools.layout.StationSlotLayout;
+import slimeknights.tconstruct.library.tools.layout.StationSlotLayoutLoader;
+import slimeknights.tconstruct.tables.client.inventory.TinkerStationScreen;
+
+import java.util.List;
+
+public class TinkerStationButtonsWidget extends SideButtonsWidget<SlotButtonItem> {
+
+  public static final int WOOD_STYLE = 2;
+  public static final int METAL_STYLE = 1;
+
+  public TinkerStationButtonsWidget(TinkerStationScreen parent, int leftPos, int topPos, int style) {
+    this(parent, leftPos, topPos, createLayoutsList(parent), style);
+  }
+
+  public TinkerStationButtonsWidget(TinkerStationScreen parent, int leftPos, int topPos, List<StationSlotLayout> layouts, int style) {
+    super(parent, leftPos, topPos, TinkerStationScreen.COLUMN_COUNT, rowsForCount(TinkerStationScreen.COLUMN_COUNT, layouts.size()),
+      SlotButtonItem.WIDTH, SlotButtonItem.HEIGHT);
+
+    // Logic to run when a button is pressed
+    Button.OnPress onButtonPressed = self -> {
+      for (SlotButtonItem button : TinkerStationButtonsWidget.this.buttons) {
+        button.pressed = false;
+      }
+      if (self instanceof SlotButtonItem slotInformationButton) {
+        slotInformationButton.pressed = true;
+        parent.onToolSelection(slotInformationButton.getLayout());
+      }
+    };
+
+    // create buttons for layouts
+    for (int index = 0; index < layouts.size(); index++) {
+      StationSlotLayout layout = layouts.get(index);
+
+      SlotButtonItem slotButtonItem = new SlotButtonItem(index, -1, -1, layout, onButtonPressed);
+      this.addInfoButton(slotButtonItem, style);
+      if (layout == parent.getCurrentLayout()) {
+        slotButtonItem.pressed = true;
+      }
+    }
+
+    this.setButtonPositions();
+  }
+
+  public static List<StationSlotLayout> createLayoutsList(TinkerStationScreen parent) {
```

Comment:
Creating this list might be better suited to be done inside `TinkerStationScreen`. Thoughts?

Reply:
given it takes the screen as a parameter, yes